website/content/docs/api-gateway/configuration/gatewayclassconfig.mdx

```diff
@@ -0,0 +1,196 @@
+
+You can specify the following strings:
+* `http`
+* `https`
+
+### copyAnnotations.service
+List of kubernetes [annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/) to copy to the gateway service.
+* Type: Array of Strings
+* Required: optional
+
+### deployment.defaultInstances
+Specifies the number of gateway instances to deploy per gateway configuration
+* Type: Integer
+* Required: optional
+* Default: 1
+
+### deployment.maxInstances
+Specifies the maximum allowed number of gateway instances per gateway configuration
+* Type: Integer
+* Required: optional
+* Default: 8
+
+### deployment.minInstances
+Specifies the minimum allowed number of gateway instances per gateway configuration
+* Type: Integer
+* Required: optional
+* Default: 1
+
+### image.consulAPIGateway
+The image to use for consul-api-gateway. View available image tags on [DockerHub](https://hub.docker.com/r/hashicorp/consul-api-gateway/tags).
+
+Most deployments will use the default value unless a specific version of the Consul API Gateway is desired.
+* Type: string
+* Required: optional
+* Default: `"hashicorp/consul-api-gateway:RELEASE_VERSION"`
+
+### image.envoy
+The image to use for consul-api-gateway. View available image tags on [DockerHub](https://hub.docker.com/r/hashicorp/consul-api-gateway/tags).
+
+Most deployments will use the default value unless a specific version of Envoy is desired.
+* Type: string
+* Required: optional
+* Default: `"envoyproxy/envoy:RELEASE_VERSION"`
+
+### logLevel
+Specifies the error reporting level for logs.
+* Type: string
+* Required: optional
+* Default: `info`
+
+You can specify the following strings:
+* `error`
+* `warning`
+* `info`
+* `debug`
+* `trace`
+
+### nodeSelector
+Specifies a set of parameters that constrain the nodes on which the pod can run. Defining nodes with the `nodeSelector` enables the pod to fit on a node. The selector must match a node's labels for the pod to be scheduled on that node. Refer to the [Kubernetes documentation](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/) for additional information.* Type: string
+* Required: optional
+
+### serviceType
+Specifies the ingress methods for a service.
+* Type: string
+* Required: optional
+
+You can specify the following strings:
+* `ClusterIP`: Gateway will only be accessible from inside the cluster
+* `NodePort`: Gateway will be exposed on each Kubernetes node at a static port
+* `LoadBalancer`: Gateway will be exposed to external traffic by a load balancer
+
+For more on Kubernetes services, see [Publishing Services](https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types)
+
+### useHostPorts
+If set to `true`, then the Envoy container ports are mapped to host ports.
+* Type: boolean
+* Required: optional
+* Default: `false`
+
+
+## Full Configuration
+The following example creates a gateway class configuration called `test-gateway-class-config`:
```

Comment:
```suggestion
The following example creates a gateway class configuration called `test-gateway-class-config`. Traffic that passes through gateways created from the class configuration authenticate with Consul over HTTPS on port `8501`. Consul client agents communicate with server agents on port `8502` :
```
Am I describing what the example does correctly? Are there any other configurations we could add that would be useful for readers? The examples are opportunities to show-and-tell a scenario.